pex/pip/version.py

```diff
@@ -257,21 +257,13 @@ def values(cls):
         requires_python=">=3.7,<3.13",
     )
 
-    # This is https://github.com/pypa/pip/pull/12462 which is approved but not yet merged or
-    # released. It allows testing Python 3.13 pre-releases but should not be used by the public; so
-    # we keep it hidden.
-    v24_0_dev0_patched = PipVersionValue(
-        name="24.0.dev0-patched",
-        version="24.0.dev0+patched",
-        requirement=(
-            "pip @ git+https://github.com/jsirois/pip@0257c9422f7bb99a6f319b54f808a5c50339be6c"
-        ),
-        setuptools_version="69.0.3",
-        wheel_version="0.42.0",
-        requires_python=">=3.7",
-        hidden=True,
+    v24_1 = PipVersionValue(
+        version="24.1",
+        setuptools_version="70.1.0",
+        wheel_version="0.43.0",
+        requires_python=">=3.8,<3.14",
     )
 
     VENDORED = v20_3_4_patched
     LATEST = LatestPipVersion()
-    DEFAULT = DefaultPipVersion(preferred=(VENDORED, v23_2))
+    DEFAULT = DefaultPipVersion(preferred=(VENDORED, v23_2, v24_1))
```

Comment:
Just to confirm by default pex will select the first compatible pip version right?

Reply:
That's right. These entries effectively mark where Python broke Pip. In this case, Pip relied on APIs in `importlib` that Python yanked in 3.13; so Pip 24.1 is the first Pip version to support `Python>=3.13`. So that is the default for `Python>=3.13` until some new Python no longer works against Pip 24.1 in which case I add a new entry. The Pip 23.2 marks the transition of Pip to Python 3.12 support where Python yanked `distutils`, breaking Pip.